### ISSUE TITLE
Problem with dates after the last reticulate update

### ISSUE THREAD

**Issue by grayskripko** [magnusfurugard/featuretoolsR#12]
"Reticulate now always converts R Date objects into Python datetime objects. Note that these conversions can be inefficient -- if you would prefer conversion to NumPy datetime64 objects / arrays, you should convert your date to POSIXct first."
https://github.com/rstudio/reticulate/blob/master/NEWS.md

The next lines take a couple of seconds on my machine with a good CPU
```
rep(as_date("2019-01-01"), 500) %>% reticulate::r_to_py()
```

I fixed it in my project with
```
r_tibble %>% mutate_if(is.Date, as.POSIXct) %>% reticulate::r_to_py()
```

You should add this kind of mutation for every incoming R data.frame

**Comment by magnusfurugard** [comment]
Good input, however, coercing to `POSIXct` also unintentionally adds a time zone to the date, which means the date potentially no longer is the same. I think a warning could be a better approach?

**Comment by grayskripko** [comment]
if the user table contains a date column with 10000+ values the R session can be treated as frozen. So, the warning doesn't actually help but definitely is much better than a quiet problem

**Comment by magnusfurugard** [comment]
I've been thinking about this issue along with a few more of the open ones. I think you're bringing up a fair point, but I don't want to forcibly change user values. On the other hand, if it makes the R session unusable we should adress it forcibly..

What do think about adding support for this through a option-toggle (default on package load: on)? 

Something along the lines of:
```
# Loads on package attach
options(
    featuretoolsR.force_posixct = T, #default
    featuretoolsR.posixct_tz = "UTC" #default
)
```

Runs something like this on entityset creation:
```
if(getOption("featuretoolsR.force_posixct")) {
  .cols = lapply(df, class)
  for (i in 1:length(.cols)) {
    colname <- names(.cols)[i]
    coltype <- .cols[[colname]]
    if (any(coltype == "Date")) {
      df[,colname] <- as.POSIXct(df[,colname], getOption("featuretoolsR.posixct_tz"))
    }
  }
}
```

**Comment by magnusfurugard** [comment]
I added something like this to the current development-branch (990b40e3eae2dd96404b818d2abec3f52b01c735). Feel free to try it out.

**Comment by grayskripko** [comment]
Sorry, I don't feel I can help here

**Comment by magnusfurugard** [comment]
Resolved in #17
